xwiki-platform-core/xwiki-platform-oldcore/src/main/java/org/xwiki/store/TemporaryAttachmentSessionsManager.java

```diff
@@ -74,6 +74,22 @@ XWikiAttachment uploadAttachment(DocumentReference documentReference, Part part)
     XWikiAttachment uploadAttachment(DocumentReference documentReference, Part part, String filename)
         throws TemporaryAttachmentException;
 
+    /**
+     * Allow to temporarily attach the given instance of {@link XWikiAttachment} to the given document reference.
+     * This can be useful if an API manipulates an {@link XWikiAttachment} without saving it, and want it to be
+     * discoverable when parsing a document through the download action for example.
+     * Note that consumer of this API needs to be aware that the file attached to the {@link XWikiAttachment} might be
+     * deleted at the end of the session, as any temporary attachment.
```

Comment:
Is there a risk that the temporary attachments are viewable by other means (until the session is ended).
For instance, the attachment picker is now listing temporary attachments, could they be displayed there?

Reply:
Yes of course, the logic of the temporary attachment session doesn't change, so it's responsibility of the consumer of the API to be careful about what he's doing and to properly call the API to delete the temporary attachment when needed.

Reply:
thanks, sounds good